### PR TITLE
8266155: Convert java.base to use Stream.toList()

### PR DESCRIPTION
**Pull request by igraves**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266155](https://bugs.openjdk.java.net/browse/JDK-8266155): Convert java.base to use Stream.toList()

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to 3fb335e8729655fce0984a6e98e22acf0a995dd3
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to 3fb335e8729655fce0984a6e98e22acf0a995dd3
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3734/head:pull/3734` \
`$ git checkout pull/3734`

Update a local copy of the PR: \
`$ git checkout pull/3734` \
`$ git pull https://git.openjdk.java.net/jdk pull/3734/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3734`

View PR using the GUI difftool: \
`$ git pr show -t 3734`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3734.diff">https://git.openjdk.java.net/jdk/pull/3734.diff</a>

</details>

### CODE REVIEW

**Review by bplb — APPROVED** [review]
Looks all right.

**Review by naotoj — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

File: src/java.base/share/classes/java/lang/invoke/MethodHandles.java [hunk 1]
```diff
@@ -6779,11 +6778,11 @@ private static void loopChecks2(List<MethodHandle> step, List<MethodHandle> pred
...
-        }).collect(Collectors.toList());
+        }).toList();
     }
 
     private static List<MethodHandle> fixArities(List<MethodHandle> hs) {
-        return hs.stream().map(MethodHandle::asFixedArity).collect(Collectors.toList());
+        return hs.stream().map(MethodHandle::asFixedArity).toList();
     }
 
     /**
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/lang/invoke/MethodHandles.java:6788]: I think the import of Collectors can be dropped in this file? And maybe a few other files too?

**Review by ChrisHegarty — APPROVED** [review]

File: src/java.base/share/classes/java/lang/invoke/MethodHandles.java (hunk 1 above)
Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/invoke/MethodHandles.java:6788]: The import can be dropped from the jdk.internal.module.* classes too.

jdk.internal.module.IllegalAccessLogger will likely be removed as part of JEP 403.

**Review by irisclark — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]

**Pull request closed without merging**